nx/lib/nx.ex

```diff
@@ -5244,6 +5244,81 @@ defmodule Nx do
     impl!(tensor).conjugate(%{tensor | type: Nx.Type.to_complex(tensor.type)}, tensor)
   end
 
+  @doc """
+  Returns the real component of each entry in a complex tensor
+  as a floating point tensor.
+
+  ## Examples
+
+      iex> Nx.real(Complex.new(1, 2))
+      #Nx.Tensor<
+        f32
+        1.0
+      >
+
+      iex> Nx.real(Nx.tensor(1))
+      #Nx.Tensor<
+        f32
+        1.0
+      >
+
+      iex> Nx.real(Nx.tensor([Complex.new(1, 2), Complex.new(2, -4)]))
+      #Nx.Tensor<
+        f32[2]
+        [1.0, 2.0]
+      >
+  """
+  @doc type: :element
+  def real(tensor) do
+    case to_tensor(tensor) do
+      %{type: {:c, size}} = tensor ->
+        impl!(tensor).real(%{tensor | type: {:f, div(size, 2)}}, tensor)
+
+      %{type: {:f, _}} = tensor ->
```

Comment:
This will do a conversion for :bf. You need to use `Nx.Type.floating?` instead. Perhaps we need a test.